lib/index.js

```diff
@@ -217,3 +282,67 @@ conversions["RegExp"] = function (V, opts) {
 
     return V;
 };
+
+conversions["Error"] = function (V, opts) {
+    if (!(V instanceof Error)) {
+        throw new TypeError("Argument is not an Error object");
+    }
+
+    return V;
+};
+
+conversions["ArrayBuffer"] = function (V, opts) {
+    // Unfortunately, the IsDetachedBuffer abstract operation is not exposed
+    // through JS.
+    if (!(V instanceof ArrayBuffer)) {
+        throw new TypeError("Argument is not an ArrayBuffer object");
+    }
+
+    return V;
+};
+
+conversions["DataView"] = function (V, opts) {
+    if (!(V instanceof DataView)) {
+        throw new TypeError("Argument is not a DataView object");
+    }
+
+    return V;
+};
+
+[
+    Int8Array, Int16Array, Int32Array, Uint8Array, Uint16Array, Uint32Array,
+    Uint8ClampedArray, Float32Array, Float64Array
+].forEach(function (func) {
+    const name = func.name;
+    conversions[name] = function (V, opts) {
+        if (!(V instanceof func)) {
+            throw new TypeError("Argument is not a " + name + " object");
+        }
+
+        return V;
+    };
+});
+
+// Common definitions
+
+conversions["ArrayBufferView"] = function (V, opts) {
+    if (!ArrayBuffer.isView(V)) {
+        throw new TypeError("Argument is not a view on an ArrayBuffer object");
+    }
+
+    return V;
+};
+
+conversions["BufferSource"] = function (V, opts) {
+    if (!(ArrayBuffer.isView(V) || V instanceof ArrayBuffer)) {
+        throw new TypeError("Argument is not an ArrayBuffer object or a view on one");
+    }
+
+    return V;
+};
+
+conversions["DOMTimeStamp"] = conversions["unsigned long long"];
+
+conversions["Function"] = convertCallbackFunction;
+
+conversions["VoidFunction"] = convertCallbackFunction;
```

Comment:
It might be nice to have the converted version always return `undefined`, regardless of what the original does. But I guess probably not worth the potential runtime overhead and extra stack frame in stack traces...